flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResult.java

```diff
@@ -171,6 +180,21 @@ boolean areAllPartitionsFinished() {
         return numberOfRunningProducers.get() == 0;
     }
 
+    public MaybeOffloaded<ShuffleDescriptor[]> getCachedShuffleDescriptors(
+            ConsumedPartitionGroup consumedPartitionGroup) {
+        return shuffleDescriptorCache.get(consumedPartitionGroup);
+    }
+
+    public void cacheShuffleDescriptors(
+            ConsumedPartitionGroup consumedPartitionGroup,
+            MaybeOffloaded<ShuffleDescriptor[]> shuffleDescriptors) {
+        this.shuffleDescriptorCache.put(consumedPartitionGroup, shuffleDescriptors);
+    }
+
+    public void notifyPartitionStatusChanged() {
+        this.shuffleDescriptorCache.clear();
```

Comment:
The offloaded data should be removed once not needed. Otherwise there can be lots of outdated blob files.
However, I'm a bit concerned that it can happen that the blob is deleted when a task still needs it.

Reply:
Thank you for pointing this out. This may requires further thinking about this.